src/main/java/edu/cornell/kfs/module/ar/service/impl/CuCustomerAddressHelperServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package edu.cornell.kfs.module.ar.service.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.integration.cg.ContractsAndGrantsBillingAward;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.ArConstants.ContractsAndGrantsInvoiceDocumentCreationProcessType;
+
+import edu.cornell.kfs.module.ar.service.CuCustomerAddressHelperService;
+
+public class CuCustomerAddressHelperServiceImpl implements CuCustomerAddressHelperService {
+    
+    private static final Logger LOG = LogManager.getLogger();
+    
+    /*
+     * CU Customization: KFSPTS-26393
+     * When creationProcessType is BATCH, validate that the customer number on the award as part if the customer
+     * address composite key matches the customer number associated to the agency associated to the award.
+     */
+    public boolean agencyCustomerMatchesAwardCustomer(ContractsAndGrantsBillingAward award,
+            ContractsAndGrantsInvoiceDocumentCreationProcessType creationProcessType) {
+        
+        if (ContractsAndGrantsInvoiceDocumentCreationProcessType.BATCH == creationProcessType) {
+            if (ObjectUtils.isNotNull(award.getAgency().getCustomer().getCustomerNumber()) &&
```

Comment:
I think StringUtils.isNotBlank is checking for null also. I wonder if this should instead check that the agency and customer objects are not null

Reply:
After Nancy tested the ObjectUtils.isNotNull method it confirmed my suggestion that it does not check on all the objects in the chain and that if agency or customer are null then it would throw a null pointer exception. I see that the code has been updated to account for this. Thank you.